src/box/iproto.cc

```diff
@@ -3461,6 +3567,43 @@ iproto_send_start_msg(void)
 		iproto_do_cfg(&iproto_threads[i], &cfg_msg);
 }
 
+void
+iproto_shutdown(void)
```

Comment:
We need a separate function to drop all existing connections for https://github.com/tarantool/tarantool-ee/issues/585. Let's implement a function that would do just that, without deinitializing iproto. Since we stop accepting new connections in `iproto_on_shutdown_f`, this should be enough to shutdown iproto gracefully - we'd just need to stop and join all iproto threads in `iproto_free`.

Reply:
Done.

```c
/**
 * Drop all current connections. That is stop IO and cancel all inprogress
 * requests. Return when the requests are finished and connection is freed.
 * Concurrent calls are serialized.
 */
void
iproto_drop_connections(void);
```